### PR TITLE
[fix](olap) The 'scan key' generated by the 'is null' expression causes incorrect query results

### LINKED ISSUE #17568: [Bug] Got incorrect result with scan keys for boolean column
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

* master
* 1.2

### What's Wrong?

1. create the table and insert data:
```sql
create table test_null (
    id boolean null,
    value int null
) duplicate key(id) 
DISTRIBUTED BY HASH(id) buckets 1 
PROPERTIES ("replication_allocation" = "tag.location.default: 1");

insert into test_null values(1, null), (1,1), (null, 2), (1,2), (0, 3), (0, 4);
```
2. check the table's rows
```sql
select * from test_null;
```
got result:
```
+------+-------+
| id   | value |
+------+-------+
| NULL |     2 |
|    0 |     4 |
|    0 |     3 |
|    1 |     2 |
|    1 |     1 |
|    1 |  NULL |
+------+-------+
```

3. test with `is null`
```sql
select * from test_null where id is null;
```
got result:
```
+------+-------+
| id   | value |
+------+-------+
| NULL |     2 |
|    0 |     4 |
|    0 |     3 |
+------+-------+
```

### What You Expected?

got result:
```
+------+-------+
| id   | value |
+------+-------+
| NULL |     2 |
+------+-------+
```

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes

Issue Number: close #17568

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

File: regression-test/suites/correctness_p0/test_null_predicate.groovy [hunk 1]
```diff
@@ -115,4 +115,21 @@ suite("test_null_predicate") {
...
+
+    sql """ DROP TABLE IF EXISTS test_null_predicate"""
+    sql """
+        create table test_null_predicate (
+            id boolean null,
+            value int null
+        ) duplicate key(id)
+        DISTRIBUTED BY HASH(id) buckets 1
+        PROPERTIES ("replication_allocation" = "tag.location.default: 1");
+    """
```
Comment (yiguolei) [inline regression-test/suites/correctness_p0/test_null_predicate.groovy:127]: value column is not the key column, the scan key does have affect on this?
I think it's better to add a case that the predicate column is nullable and it is also part of duplcate column.

File: regression-test/suites/correctness_p0/test_null_predicate.groovy [hunk 2]
```diff
@@ -115,4 +115,21 @@ suite("test_null_predicate") {
...
 
     qt_select15 """ select * from test_null_predicate2 where `value` is null order by id; """
+
+    sql """ DROP TABLE IF EXISTS test_null_predicate"""
+    sql """
+        create table test_null_predicate (
+            id boolean null,
+            value int null
+        ) duplicate key(id)
+        DISTRIBUTED BY HASH(id) buckets 1
```
Comment (yiguolei) [inline regression-test/suites/correctness_p0/test_null_predicate.groovy:125]: set disable_auto_compaction to true. Or the data may be compacted and the case not stable.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.23 seconds
 stream load tsv:          452 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          74 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230308112435_clickbench_pr_110773.html

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**